mkosi/resources/mkosi-obs/mkosi.build

```diff
@@ -129,4 +129,36 @@ if [ -d hashes/ukis ]; then
     echo "Staging the following files for signing:"
     cpio -t <"$OUTPUTDIR/hashes.cpio.rsasign"
 fi
+
+# Fifth step: finalize any DDI by attaching the verity roothash signatures
+while read -r SIG; do
+    test -f "/usr/src/packages/SOURCES/$(basename "${SIG%roothash.sig}repart.tar")" || continue
+
+    PATH=/usr/lib/systemd/:$PATH systemd-keyutil --help | grep -q 'pkcs1-to-pkcs7' || break
+    PATH=/usr/lib/systemd/:$PATH systemd-keyutil pkcs1-to-pkcs7 --certificate /usr/src/packages/SOURCES/_projectcert.crt --output "${SIG%sig}p7s" "$SIG"
+
+    mkdir -p mkosi.repart
+    tar xf "/usr/src/packages/SOURCES/$(basename "${SIG%roothash.sig}repart.tar")" -C mkosi.repart
+    rm -f "$OUTPUTDIR/$(basename "${SIG%roothash.sig}repart.tar")"
+    if [ -f "$OUTPUTDIR/$(basename "${SIG%roothash.sig}raw.zst")" ]; then
+        recompress=yes
+        unzstd "$OUTPUTDIR/$(basename "${SIG%roothash.sig}raw.zst")"
+    fi
+    # sd-repart might be configured to create labels based on os-release fields. Due to the
+    # mkosi sandbox we cannot use --image, so we need to provide the os-release file manually
+    # via --root
+    if [ -f "/usr/src/packages/SOURCES/$(basename "${SIG%roothash.sig}osrelease")" ]; then
+        mkdir -p mkosi.repart/usr/lib
+        cp "/usr/src/packages/SOURCES/$(basename "${SIG%roothash.sig}osrelease")" mkosi.repart/usr/lib/os-release
+        root_arg="--root=$PWD/mkosi.repart"
+    fi
+    systemd-repart "${root_arg:-}" --join-signature "$(cat "${SIG%.sig}"):${SIG%sig}p7s" --certificate /usr/src/packages/SOURCES/_projectcert.crt --definitions mkosi.repart --dry-run=no "$OUTPUTDIR/$(basename "${SIG%roothash.sig}raw")"
```

Comment:
When not exchanging `root_arg` for anything, the `:-` is not needed, also, with the quoting around it, if `root_arg` is empty, repart will be passed an empty string as argument, without the quoting, no argument is passed in that case
```suggestion
    systemd-repart $root_arg --join-signature "$(cat "${SIG%.sig}"):${SIG%sig}p7s" --certificate /usr/src/packages/SOURCES/_projectcert.crt --definitions mkosi.repart --dry-run=no "$OUTPUTDIR/$(basename "${SIG%roothash.sig}raw")"
```

Reply:
In mkosi/resources/mkosi-obs/mkosi.build line 156:
        $root_arg \
        ^-------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Reply:
shellcheck is not gospel. :) It can't know whether expanding there is sensible or not and this would probably be a good point to put a `# shellcheck disable=SC2086` line above this one.